app/gui2/src/stores/graph/index.ts

```diff
@@ -304,6 +316,34 @@ export const useGraphStore = defineStore('graph', () => {
     editedNodeInfo.value = { id, initialCursorPos: cursorPosition }
   }
 
+  function getPortPrimaryInstance(id: PortId): PortViewInstance | undefined {
+    const instances = portInstances.get(id)
+    return instances && set.first(instances)
+  }
+
+  /**
+   * Get the bounding rectangle of a port view, within the coordinate system of the node it belongs
+   * to. If the port is currently not connected or interacted with, `undefined` may be returned.
+   */
+  function getPortRelativeRect(id: PortId): Rect | undefined {
+    return getPortPrimaryInstance(id)?.rect.value
+  }
+
+  function getPortNodeId(id: PortId): ExprId | undefined {
+    return getPortPrimaryInstance(id)?.nodeId ?? db.getExpressionNodeId(id as string as ExprId)
+  }
+
+  /**
+   * Emit an value update to a port view under specific ID. Returns `true` if the port view is
+   * registered and the update was emitted, or `false` otherwise.
+   */
+  function updatePortValue(id: PortId, value: Ast.Ast | undefined): boolean {
+    const update = getPortPrimaryInstance(id)?.onUpdate
```

Comment:
The `onUpdate` prop is equivalent to `update` emit, and can be used with `@update` notation in the template. Since it is not possible to nicely define emit types in the generic component instnace, I opted to use the prop notation to do the same thing.

Reply:
> The onUpdate prop is equivalent to update emit, and can be used with @update notation in the template.

So this is my entire point. If I see `@update` I assume this is something what is emitted after/during update I can connect to - not something I need to _call_ to do actual update, a here.